examples/sync-merkleroots.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  ErrorNoAdminKey,
+  ErrorResponse,
+  MerkleRoot,
+  Repository,
+  SpvWalletClient,
+  SpvWalletError,
+} from '../dist/typescript-npm-package.cjs.js';
+import { exampleXPub } from './example-keys.js';
+import { errMessage } from './utils.js';
+
+const server = 'http://localhost:3003/api/v1';
+
+// perhaps change exmapleXPub to actual existing in db
+if (!exampleXPub) {
+  console.log(errMessage('xPub'));
+  process.exit(1);
+}
+
+const client = new SpvWalletClient(server, {
+  xPub: exampleXPub,
+});
+
+// simulate a storage of merkle roots that exists on a client side that is using SyncMerkleRoots method
+const db: {
+  merkleRoots: MerkleRoot[];
+} = {
+  merkleRoots: [
+    {
+      merkleRoot: '4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b',
+      blockHeight: 0,
+    },
+    {
+      merkleRoot: '0e3e2357e806b6cdb1f70b54c3a3a17b6714ee1f0e68bebb44a74b1efd512098',
+      blockHeight: 1,
+    },
+    {
+      merkleRoot: '9b0fc92260312ce44e74ef369f5c66bbb85848f2eddd5a7a1cde251e54ccfdd5',
+      blockHeight: 2,
+    },
+  ],
+};
+
+// simulate repository
+const repository: Repository = {
+  saveMerkleRoots: (syncedMerkleRoots: MerkleRoot[]) => {
+    console.log('\nsaveMerkleRoots called\n');
+    db.merkleRoots.push(...syncedMerkleRoots);
+
+    return new Promise((resolve) => setTimeout(resolve, 1000));
+  },
+  getLastEvaluatedKey: () => {
+    return new Promise((resolve) => {
+      if (db.merkleRoots.length < 1) {
+        resolve(undefined);
+      } else {
+        resolve(db.merkleRoots[db.merkleRoots.length - 1].merkleRoot);
+      }
+    });
+  },
+};
+
+try {
+  console.log('INITIAL STATE: ', db.merkleRoots.length, '\n\n');
+
+  await client.SyncMerkleRoots(repository);
+
+  console.log('AFTER SYNC', db.merkleRoots.length, '\n\n');
```

Comment:
Maybe, but... maybe - we could print the last or a few of the last items? 🤔

Reply:
done